### ISSUE TITLE
Error message from FindMarkers() when using assays="SCT"

### ISSUE THREAD

**Issue by weishwu** [satijalab/seurat#5823]
Seurat version: 4.1.0
My command-lines:
```
# Data from different samples were normalized using SCTransform and then integrated. And I did run PrepSCTFindMarkers on vs_integrated_cluster.
vs_integrated_cluster <- PrepSCTFindMarkers(vs_integrated_cluster)
vs_integrated_cluster_EGFP <- subset(vs_integrated_cluster, EGFP > 0)
FindMarkers(vs_integrated_cluster_EGFP, ident.1 = "0", ident.2 = "1", verbose = FALSE, logfc.threshold=-Inf, min.pct=0.1, assay = 'SCT', slot = 'data')
```
The FindMarkers() line gave an error:
```
Error in if (any(unlist(model_median_umis) != min_median_umi)) { : 
  missing value where TRUE/FALSE needed
```

Running FindMarkers() on the entire set vs_integrated_cluster worked.

**Comment by saketkc** [comment]
You will need to switch to `SCT` assay and run `PrepSCTFindMarkers` for performing DE on SCT assay (if using the v2 model). An example is in [this vignette](https://satijalab.org/seurat/articles/sctransform_v2_vignette.html)

**Comment by weishwu** [comment]
> You will need to switch to `SCT` assay and run `PrepSCTFindMarkers` for performing DE on SCT assay (if using the v2 model). An example is in [this vignette](https://satijalab.org/seurat/articles/sctransform_v2_vignette.html)

@saketkc 
Thanks for the reply.
But I already ran `PrepSCTFindMarkers` on the full set (vs_integrated_cluster) and as I said running FindMarkers() on the full set worked. The error only happened when I ran FindMarkers() on the subset (vs_integrated_cluster_EGFP). I also tried running `PrepSCTFindMarkers` again (which I don't think is needed though) on vs_integrated_cluster_EGFP but it gave the same error. Also `DefaultAssay(vs_integrated_cluster_EGFP)` gave "SCT".
